controllers/keycloakclient_reconciler.go

```diff
@@ -26,6 +34,38 @@ func NewDedicatedKeycloakClientReconciler(keycloak kc.Keycloak) *DedicatedKeyclo
 		Keycloak: keycloak,
 	}
 }
+func GetClientShaCode(clientID string) (string, error) {
+	secretSeed, _ := getSecretSeed()
+	if secretSeed != "" {
+		h := sha256.New()
+		h.Write([]byte(secretSeed + clientID + model.SALT))
```

Comment:
Woher kommt der secretSeed? Ist der pro Keyckloak-Installation oder pro Client unique und ersetzt der secretSeed damit nicht schon den SALT?

Reply:
secretSeed ist der eigentliche Schutz, den gibt es nur einmal pro Installation des Controllers und wird mit Keycloak Mitteln geschützt.
Damit jeder Client einen einndeutigen Schlüssel hat, kommt der Client mit in die Sha Generierung rein.

Reply:
Der Salt soll vor der Verwendung vorgenerierter Rainbow Tables schützen. Ein statischer Salt kann davor tatsächlich schützen. Im konkreten Fall verwendest du mit `secretSeed` einen dynamischen, pro Installation, eindeutigen Salt. Damit ist der statische "model.SALT" überflüssig (macht jetzt aber auch nichts "kaputt").